### PR TITLE
Added in Debug Rendering as a BFS to Render Depth

### LINKED ISSUE #21: Update the Debug Colours to Show Nested Depth
Currently the debug flag will paint the view and it's children in separate colours. It would be cool if there were different colours that shows the depth of nested children.

### PR DESCRIPTION
**Pull request by OpenSauce**
Hello! 

Closes #21 This was an interesting one, took me a while to figure out a nice way of doing it but I managed it in the end.

I've added in a bit more to the debug, it now renders using a BFS so that I can track the depth of the levels of children easily.

If there's any issues/things I can improve let me know!

### CODE REVIEW

**Comment by yohamta** [comment]
@OpenSauce Hello!
Wow, it's really amazing! Thank you very much. It will help us a lot for debugging :)

File: view.go [hunk 1]
```diff
@@ -62,13 +62,16 @@ func (v *View) UpdateWithSize(width, height int) {
...
 	}
-
 	v.Update()
 }
 
 // Draw draws the view
 func (v *View) Draw(screen *ebiten.Image) {
 	v.containerEmbed.Draw(screen)
+
+	if Debug {
```
Comment (yohamta) [inline view.go:72]: Should we change the if condition as below? I thought this way we can reduce the draw count for performance :) What do you think?
```go
	if Debug && !v.hasParent {
		debugBorders(screen, v.containerEmbed)
	}
```

Comment (OpenSauce) [reply view.go]: Yep, 100%, just updated this

**Comment by yohamta** [comment]
@OpenSauce Thank you very much! It is tagged as `v2.1.7`.

**Pull request merged**